### PR TITLE
Add auth header define in envvars

### PR DESCRIPTION
**Pull request by psiinon**
Will def need to document this ;)

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: zap/src/main/java/org/zaproxy/zap/extension/authentication/HttpSenderAuthHeaderListener.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+class HttpSenderAuthHeaderListener implements HttpSenderListener {
+
+    private final String authHeaderValue;
+    private final String authHeader;
+    private final String authHeaderSite;
+
+    public HttpSenderAuthHeaderListener() {
+        authHeaderValue = System.getenv("ZAP_AUTH_HEADER_VALUE");
+        String authHeaderVar = System.getenv("ZAP_AUTH_HEADER");
+        if (authHeaderVar == null) {
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/authentication/HttpSenderAuthHeaderListener.java:36]: or empty?

**Comment by thc202** [comment]
Could you add tests?

**Comment by psiinon** [comment]
Tweaked and added tests.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thanks!

File: zap/src/main/java/org/zaproxy/zap/extension/authentication/HttpSenderAuthHeaderListener.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    public static final String ZAP_AUTH_HEADER_VALUE = "ZAP_AUTH_HEADER_VALUE";
+    public static final String ZAP_AUTH_HEADER = "ZAP_AUTH_HEADER";
+    public static final String ZAP_AUTH_HEADER_SITE = "ZAP_AUTH_HEADER_SITE";
+
+    private final String authHeaderValue;
+    private final String authHeader;
+    private final String authHeaderSite;
+
+    public HttpSenderAuthHeaderListener(Function<String, String> propertyProvider) {
```
Comment (ricekot) [inline zap/src/main/java/org/zaproxy/zap/extension/authentication/HttpSenderAuthHeaderListener.java:38]: Wow, I was not aware we could pass functions as parameters, that's so cool!

**Comment by ricekot** [comment]
I really enjoyed reading the code from this PR 😃. Thank you!

**Review by ricekot — APPROVED** [review]

**Pull request merged**